lib/Doctrine/DBAL/Connection.php

```diff
@@ -1589,11 +1589,17 @@ public function ping()
             return $this->_conn->ping();
         }
 
+        set_error_handler(function ($severity, $message) {
+            throw new \PDOException($message, $severity);
```

Comment:
I was gonna implement this in `PDOMySQL` but I wasn't sure about the performance impact of attaching and reseting an error handler on every query.